Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitUserContentManager.cpp

```diff
@@ -279,8 +279,7 @@ class UserScriptMessageTest : public WebViewTest {
 
     bool registerHandler(const char* handlerName, const char* worldName = nullptr)
     {
-        return worldName ? webkit_user_content_manager_register_script_message_handler_in_world(m_userContentManager.get(), handlerName, worldName)
-            : webkit_user_content_manager_register_script_message_handler(m_userContentManager.get(), handlerName);
+        return webkit_user_content_manager_register_script_message_handler_with_reply(m_userContentManager.get(), handlerName, worldName);
```

Comment:
While technically this is okay, if we replace the existing implementation we no longer have a test for the old functions. We should add instead a `registerAsyncHandler()` function, or add a parameter to the existing one, for example:

```cpp
bool registerHandler(const char* handlerName, const char* worldName, bool async)
{
    if (async) {
        // Use the new function.
    } else {
        // The existing code here goes here, so we can continue testing the old functions.
    }
}